lib/vmdb/loggers.rb

```diff
@@ -129,6 +129,10 @@ def self.contents(log, last = 1000)
       # after opening it as binary and getting rid of the invalid UTF8 byte sequences
       results.join("\n").force_encoding("utf-8")
     end
+
+    private_class_method def self.log_file_to_prog_name(log_file)
+      log_file.chomp(".log") if log_file.kind_of?(String)
```

Comment:
What is log_file when not a String?  Nil?  Anything else?  If it's just nil, does it make sense to do a guard clause:

```ruby
return if log_file.nil?

log_file.chomp(".log")
```

Reply:
It looks like we do this same logic in two places already:
```
      log_file = Pathname.new(log_file) if log_file.kind_of?(String)
      log_file = ManageIQ.root.join("log", log_file) if log_file.try(:dirname).to_s == "."
      progname = log_file.try(:basename, ".*").to_s
```

If I move the log_file => Rails.root Pathname up into the base `.create_logger` then I can assume I have a Pathname or an IO.